packages/components/src/Autocomplete/Menu/Menu.tsx

```diff
@@ -21,6 +22,25 @@ export function Menu<
   inputRef,
   customRenderMenu,
 }: MenuProps<GenericOption, GenericOptionValue>) {
+  const { open } = useModalContext();
+
+  /**
+   * Experimental/temporary workaround for Autocompletes within Modals. This is only necessary
+   * when an Autocomplete uses `customRenderMenu` and is rendered within the composable version
+   * of Modal (aka Modal.Provider).
+   *
+   * If `customRenderMenu` contains clickable elements such as Buttons, the consumer must ALSO
+   * replace any `onClick` handlers with `onMouseDown` handlers on those Buttons.
+   *
+   * The check below prevents Autocomplete from rendering the `customRenderMenu` when it's not visible
+   * (when the input isn't focused). This prevents Modals (FloatingUI) from marking the Autocomplete's
+   * menu as data-floating-ui-inert and aria-hidden="true". As a result, this prevents the bug where clicking
+   * within its menu would cause FloatingUI to close the parent Modal because it determined the click was
+   * outside of the Modal.
+   */
+  const specialModalWorkaround = open && customRenderMenu;
+  if (specialModalWorkaround && (!inputFocused || !options.length)) return null;
```

Comment:
we're going to have an issue here with `options.length`

you could have a search term of `something that doesn't match an option` so you'll have `options.length` of 0

but lets say one of your menu footer items is `Add your search term as a new option`- we wouldn't be rendering it 

<img width="430" alt="image" src="https://github.com/user-attachments/assets/68340d7f-d55c-43d7-8c5c-68dc03a7365b" />

<img width="422" alt="image" src="https://github.com/user-attachments/assets/2ac4d9b3-d65e-40cb-bef5-cbff4ff62e03" />